api/src/main/kotlin/com/google/devtools/ksp/processing/SymbolProcessorProvider.kt

```diff
@@ -3,19 +3,30 @@ package com.google.devtools.ksp.processing
 /**
  * [SymbolProcessorProvider] is the interface used by plugins to integrate into Kotlin Symbol Processing.
  */
-fun interface SymbolProcessorProvider {
+interface SymbolProcessorProvider {
     /**
-    * Called by Kotlin Symbol Processing to create the processor.
-    *
-    * @param options passed from command line, Gradle, etc.
-    * @param kotlinVersion language version of compilation environment.
-    * @param codeGenerator creates managed files.
-    * @param logger for logging to build output.
-    */
+     * Called by Kotlin Symbol Processing to create the processor.
+     *
+     * @param options passed from command line, Gradle, etc.
+     * @param kotlinVersion language version of compilation environment.
+     * @param codeGenerator creates managed files.
+     * @param logger for logging to build output.
+     */
+    @Suppress("DeprecatedCallableAddReplaceWith")
+    @Deprecated(message = "Use create that takes an environment instead")
```

Comment:
Please feel free to remove this and keep the fun interface; Backward compatibility don't need to be considered yet.

Reply:
Was wondering about that. Does this need to be a `fun interface`? It isn't used anywhere as such.

Reply:
It's fine to remove the fun interface. Can you remove the deprecated `create` and do a rebase?